app/src/main/java/org/p2p/wallet/auth/ui/phone/CountryCodeInteractor.kt

```diff
@@ -0,0 +1,64 @@
+package org.p2p.wallet.auth.ui.phone
+
+import android.content.Context
+import android.content.res.Resources
+import android.telephony.TelephonyManager
+import io.michaelrocks.libphonenumber.android.PhoneNumberUtil
+import org.p2p.wallet.auth.model.CountryCode
+import org.p2p.wallet.auth.repository.CountryCodeLocalRepository
+
+class CountryCodeInteractor(
+    private val countryCodeLocalRepository: CountryCodeLocalRepository,
+    private val phoneNumberUtil: PhoneNumberUtil
+) {
+
+    suspend fun detectCountryCodeByLocale(resources: Resources): CountryCode? {
+        return try {
+            val localeCountryIso = resources.configuration.locale.country
+            getCountryForISO(localeCountryIso)
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    suspend fun detectCountryCodeBySimCard(context: Context): CountryCode? {
+        return try {
+            val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+            val simCountryISO = telephonyManager.simCountryIso
+            getCountryForISO(simCountryISO)
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    suspend fun detectCountryCodeByNetwork(context: Context): CountryCode? {
+        return try {
+            val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
```

Comment:
we can use extension context.getSystemService<TelephonyManager> @gslevinkov am i right?

Reply:
right